docs/content/guides/dagster/code-references.mdx

```diff
@@ -0,0 +1,120 @@
+---
+title: Linking to assets' code with code references | Dagster Docs
+---
+
+# Linking to assets' code with code references
+
+<Note>
+  This feature is considered <strong>experimental</strong> and is under active
+  development. This guide will be updated as we roll out new features.
+</Note>
+
+Attaching code reference metadata to your Dagster assets allows you to easily view those assets' source code from the Dagster UI:
+
+- **In local development**, navigate directly to the code backing an asset in your editor
+- **In Dagster+**, link to source code in your source control repository to see the full change history for an asset
+
+In this guide, we'll show you how to automatically and manually attach code references to your Dagster assets.
+
+---
+
+## Prerequisites
+
+To complete the steps in this guide, you'll need:
+
+- A Dagster repository with assets that you want to link to code
+- Dagster version `1.7.6` or newer
+
+---
+
+## Automatically attaching code references to asset definitions
+
+To automatically attach code references to the asset's function definition, you can use the `dagster._core.definitions.metadata.with_source_code_references` utility. Any asset definitions passed to the utility will have their source file attached as metadata.
+
+```python file=/guides/dagster/code_references/with_source_code_references.py
+from dagster import Definitions, asset
+from dagster._core.definitions.metadata import with_source_code_references
+
+
+@asset
+def my_asset(): ...
+
+
+@asset
+def another_asset(): ...
+
+
+defs = Definitions(
+    assets=with_source_code_references(
+        [
+            my_asset,
+            another_asset,
+        ]
+    )
+)
+```
+
+A link to the asset's source in `with_source_code_references.py` will then be visible in the **Asset Catalog** view in the Dagster UI:
+
+<center>
+  <Image
+    alt="Asset catalog view showing link to with_source_code_references.py"
+    src="/images/guides/code-references/with_source_code_references.png"
+    width={1590}
+    height={326}
+  />
+</center>
+
+---
+
+## Manually attaching code references to asset definitions
+
+In some cases, you may want to manually attach code references to your asset definitions. Some assets may have a more complex source structure, such as an asset whose definition is spread across multiple Python source files or an asset which is partially defined with a `.sql` model file.
+
+To manually attach code references to an asset definition, use an instance of `dagster._core.definitions.metadata.CodeReferencesMetadataSet`. You can then choose to augment these manual references with `with_source_code_references`:
```

Comment:
I think we're not ready yet to talk about `MetadataSet`s publicly.  We should just recommend
```python
metadata={
    "dagster/code_references": CodeReferencesMetadataValue(

...
```

similar to what we do in the column lineage guide: https://docs.dagster.io/concepts/metadata-tags/asset-metadata/column-level-lineage#enabling-column-level-lineage